daemon/emer-network-send-provider.c

```diff
@@ -191,15 +191,19 @@ reset_network_send_data (EmerNetworkSendProvider *self)
                   error->message);
       g_error_free (error);
     }
+
+  priv->send_number = 0;
+  priv->data_cached = TRUE;
 }
 
-static gboolean
+static void
 read_network_send_data (EmerNetworkSendProvider *self)
 {
   EmerNetworkSendProviderPrivate *priv =
     emer_network_send_provider_get_instance_private (self);
+
   if (priv->data_cached)
```

Comment:
In most circumstances there is no guarantee that data of any type is initialized in C. gcc automatically initializes static variables to their default values (e.g., NULL, 0).

I have a vague memory of @ptomato mentioning that `g_object_new` initializes all variables in the private struct to their default values unless they are otherwise specified, but [its documentation](https://developer.gnome.org/gobject/stable/gobject-The-Base-Object-Type.html#g-object-new) only claims this about construction parameters. I wonder if the only reason that it works in practice is that the kernel initializes new pages to 0 when we fire up a new process.